lib/hdpublickey.js

```diff
@@ -65,6 +65,22 @@ function HDPublicKey(arg) {
   }
 }
 
+
+/**
+ * Verifies that a given path is valid.
+ *
+ * @param {string|number} arg
+ * @return {boolean}
+ */
+HDPublicKey.prototype.isValidPath = function(arg) {
```

Comment:
The cache is a good point. I think it's fine for now, however we may want to revisit it later. And have a method that is used to validate the path that is the same that is used for validation within the derive method.